domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1596,91 +1595,207 @@ public TotaledResults(ResultSet set, Integer total) {
 
-    vertx.runOnContext(v1 -> {
-      try {
-        QueryHelper queryHelper = buildQueryHelper(transactionMode, table, fieldName, wrapper, returnIdField, facets,
-            distinctOn);
+  /**
+   * Stream GET with CQLWrapper, no facets {@link org.folio.rest.persist.PostgresClientStreamResult}
+   * @param <T>
+   * @param table
+   * @param clazz
+   * @param fieldName
+   * @param filter
+   * @param returnIdField
+   * @param distinctOn may be null
+   * @param replyHandler AsyncResult; on success with result {@link org.folio.rest.persist.PostgresClientStreamResult}
+   */
+  public <T> void streamGet(String table, Class<T> clazz, String fieldName,
+    CQLWrapper filter, boolean returnIdField, String distinctOn,
+    Handler<AsyncResult<PostgresClientStreamResult>> replyHandler) {
 
-        connection.queryStream(queryHelper.selectQuery, stream -> {
-          if (stream.succeeded()) {
+    streamGet(table, clazz, fieldName, filter, returnIdField, distinctOn,
+      Collections.emptyList(), replyHandler);
+  }
 
-            SQLRowStream sqlRowStream = stream.result();
+  /**
+   * Stream GET with CQLWrapper and facets {@link org.folio.rest.persist.PostgresClientStreamResult}
+   * @param <T>
+   * @param table
+   * @param clazz
+   * @param fieldName
+   * @param filter
+   * @param returnIdField must be true if facets are in passed
+   * @param distinctOn may be null
+   * @param facets may not be null (Collections.emptyList() for no facets)
+   * @param replyHandler AsyncResult; on success with result {@link org.folio.rest.persist.PostgresClientStreamResult}
+   */
+  @SuppressWarnings({"unchecked", "squid:S00107"})    // Method has >7 parameters
+  public <T> void streamGet(String table, Class<T> clazz, String fieldName,
+    CQLWrapper filter, boolean returnIdField, String distinctOn,
+    List<FacetField> facets, Handler<AsyncResult<PostgresClientStreamResult>> replyHandler) {
 
-            ResultsHelper<T> resultsHelper = new ResultsHelper<>(sqlRowStream, clazz);
+    client.getConnection(conn ->
+      doStreamGet(conn, table, clazz, fieldName, filter, returnIdField,
+        distinctOn, facets, closeAndHandleResult(conn, replyHandler)));
+  }
 
-            boolean isAuditFlavored = isAuditFlavored(resultsHelper.clazz);
+  /**
+   * private for now, might be public later (and renamed)
+   * @param <T>
+   * @param connResult
+   * @param table
+   * @param clazz
+   * @param fieldName
+   * @param wrapper
+   * @param returnIdField
+   * @param distinctOn
+   * @param facets
+   * @param replyHandler
+   */
+  @SuppressWarnings({"unchecked", "squid:S00107"})    // Method has >7 parameters
+  private <T> void doStreamGet(AsyncResult<SQLConnection> connResult,
+    String table, Class<T> clazz, String fieldName, CQLWrapper wrapper,
+    boolean returnIdField, String distinctOn, List<FacetField> facets,
+    Handler<AsyncResult<PostgresClientStreamResult>> replyHandler) {
```

Comment:
Can you replace `<PostgresClientStreamResult>` by `<PostgresClientStreamResult<T>>`?